src/main/java/dev/pgm/community/users/UserProfileImpl.java

```diff
@@ -54,20 +41,18 @@ public Instant getFirstLogin() {
   }
 
   @Override
-  public Instant getLastLogin() {
-    return lastLogin;
+  public CompletableFuture<Session> getLatestSession(boolean ignoreDisguisedSessions) {
+    SessionFeature sessions = Community.get().getFeatures().getSessions();
+    if (sessions == null) return null; // this shouldn't happen
```

Comment:
Surely this shouldn't happen, but you will create a NPE on any calling code, return an instant completable future with the null if anything, but not a null directly

Reply:
```suggestion
    if (sessions == null) return  CompletableFuture.completedFuture(null);
```

Reply:
This seems like breaking error, why not throw an exception here?